File: app/src/main/java/com/playground/database/DateConverter.java
package com.playground.database;

import android.arch.persistence.room.TypeConverter;

import java.sql.Date;

/**
 * Created by emil.ivanov on 9/2/18.
 */
public class DateConverter {

    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/mainKotlin/kotlin/com.playground/data/database/DateConverter.kt
package com.playground.data.database

import android.arch.persistence.room.TypeConverter

import java.sql.Date

/**
 * Created by emil.ivanov on 9/2/18.
 */
class DateConverterKt {

    @TypeConverter
    fun toDate(timestamp: Long): Date? {
        return if (timestamp == null) null else Date(timestamp)
    }

    @TypeConverter
    fun toTimestamp(date: Date): Long? {
        return if (date == null) null else date.time
    }
}
